extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/Utils.java

```diff
@@ -214,4 +221,44 @@ public static void sendTelemetry(JavaLanguageClient client, Object message) {
     client.sendNotification(new ExecuteCommandParams("_java.gradle.buildServer.sendTelemetry",
         Arrays.asList(message)));
   }
+
+  /**
+   * Extracts the jar file from the given artifact.
+  */
+  public static File getJarFile(File file) {
```

Comment:
Question:

Do we need to do the extraction when the jar already existed?

If yes, will it overwrite the existing jar?

Reply:
If such a jar already exists then it will get overwritten by the classes.jar we are extracting. In case this is not intended, what would you suggest doing - rename the file with an index suffix (eg: library_1.jar, library_2.jar) or add a timestamp?

Reply:
Because if the dependency aar is downloaded from an open resgistry, loke gradle portal, the content of the arr and jar inside should be the same, which means we can slip extraction if jar exist.

But one concern is that for some self-build aar, the content may change but the version number may not. (Imagine a local build scenario)

Always overwrite is a safe approach, I'm just wondering if we can improve based on that

Reply:
Okay. Btw, we are extracting the JARs from the AAR file to provide JDT.LS. But this implementation is not part of the Gradle. And those extracted JARs will always sit there. I don't have much experience with vscode extensions, but if the extension has a cache directory then shouldn't we place the extracted JARs there?

Reply:
Good point, the here is how you can locate the root of the workspace cache: org.eclipse.core.resources.ResourcesPlugin.getWorkspace().getRoot().getLocation()

Reply:
And is there any limit on the cache? Saving all the extracted jars in the cache will take significant amount of space depending on the libraries being used. And presence of all these files are necessary for the entire duration the workspace is open.

Reply:
To address the space issue, we need to put it in a global location, instead of the workspace cache. Like the the current implementation, put it to the folder of gradle.

We can ask Donat about this to see if it's acceptable.

Reply:
Okay. Then I think we don't need any changes to this at the moment.